corehq/apps/sso/tests/test_decorators.py

```diff
@@ -0,0 +1,56 @@
+from unittest import mock
+
+from django.test import TestCase, RequestFactory
+
+from corehq.apps.sso.decorators import (
+    identity_provider_required,
+    use_saml2_auth,
+)
+from corehq.apps.sso.tests import generator
+
+
+class TestDecorators(TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.account = generator.get_billing_account_for_idp()
```

Comment:
If you want to break precedent, it looks like the 'correct' way to handle this is with [setUpTestData](https://docs.djangoproject.com/en/3.1/topics/testing/tools/#django.test.TestCase.setUpTestData). The advantages seem to be:
- don't have to call `super()`, because TestCase doesn't implement `setUpTestData`
- will be called once per class if the database supports transactions, otherwise will occur for every test

No 'cleanup' method exists, because the transaction takes care of the cleanup automatically for you.

It also seems setUpTestData is probably safer in the event that your code throws an exception. [This link](https://docs.djangoproject.com/en/3.1/topics/testing/tools/#django.test.SimpleTestCase.databases) claims that that can result in a seg fault on OS X.